src/gui/symbology-ng/qgsrendererv2propertiesdialog.cpp

```diff
@@ -222,7 +222,6 @@ void QgsRendererV2PropertiesDialog::rendererChanged()
     {
       if ( mMapCanvas )
         mActiveWidget->setMapCanvas( mMapCanvas );
-      changeOrderBy( mActiveWidget->renderer()->orderBy(), mActiveWidget->renderer()->orderByEnabled() );
```

Comment:
Just keep the settings in the dialog unchanged, they are what we want. The settings within the `renderer()` are only updated when applying the changes to the layer.